cpp/src/ice2slice/Gen.cpp

```diff
@@ -299,28 +299,114 @@ namespace
-                    if (q != parameterDocs.end())
+                    out << nl << "/// @param " << param->name() << ": ";
+                    for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
                     {
-                        out << nl << "/// @param " << param->name() << ": ";
-                        for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
+                        if (r != q->second.begin())
                         {
-                            if (r != q->second.begin())
-                            {
-                                out << nl;
-                            }
-                            out << (*r);
-                            r++;
+                            out << nl;
+                            out << "/// ";
                         }
+                        out << (*r);
+                        r++;
                     }
                 }
             }
+
+            // Output parameters
+            parameters = operation->outParameters();
+            for (ParameterList::const_iterator it = parameters.begin(); it != parameters.end(); it++)
+            {
+                ParameterPtr param = *it;
+                std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
+                if (q != parameterDocs.end())
+                {
+                    out << nl << "/// @returns " << param->name() << ": ";
+                    for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
+                    {
+                        if (r != q->second.begin())
+                        {
+                            out << nl;
+                            out << "/// ";
+                        }
+                        out << (*r);
+                        r++;
+                    }
+                }
+            }
+
+            // Return value
+            StringList returnDocs = comment->returns();
+            if (!returnDocs.empty())
+            {
+                out << nl << "/// @returns";
+                if (operation->returnsMultipleValues())
+                {
+                    out << " returnValue";
+                }
+                out << ": ";
+
+                for (StringList::const_iterator r = returnDocs.begin(); r != returnDocs.end();)
+                {
+                    if (r != returnDocs.begin())
+                    {
+                        out << nl;
+                        out << "/// ";
+                    }
+                    out << (*r);
+                    r++;
+                }
+            }
+
+            // Exceptions
+            for (const auto& [name, docs] : comment->exceptions())
+            {
+                out << nl << "/// @throws " << name << ": ";
+                for (StringList::const_iterator r = docs.begin(); r != docs.end();)
```

Comment:
Would be nicer if we could avoid this repeating this pattern over and over. Maybe add a helper?

Reply:
Fixed